From e3a4ae3c3700ca7150768f2d58c0b41e6d25e593 Mon Sep 17 00:00:00 2001
From: WangJing <wangjing@mprc.pku.edu.cn>
Date: Wed, 2 Nov 2011 16:06:50 +0800
Subject: [PATCH 063/641] UniCore64: Fix warning of function pte_offset_map

Fixed warnings:
mm/memory.c: In function 'copy_pte_range':
mm/memory.c:941:12: warning: initialization makes pointer from integer without a cast
mm/memory.c:944:10: warning: assignment makes pointer from integer without a cast
mm/memory.c: In function 'zap_pte_range':
mm/memory.c:1121:14: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'follow_page':
mm/memory.c:1489:9: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function '__get_user_pages':
mm/memory.c:1672:8: warning: assignment makes pointer from integer without a cast
mm/memory.c: In function '__get_locked_pte':
mm/memory.c:1971:11: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'remap_pte_range':
mm/memory.c:2166:8: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'apply_to_pte_range':
mm/memory.c:2306:3: warning: pointer/integer type mismatch in conditional expression
mm/memory.c:2307:3: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'do_wp_page':
mm/memory.c:2501:17: warning: initialization makes pointer from integer without a cast
mm/memory.c:2570:17: warning: initialization makes pointer from integer without a cast
mm/memory.c:2657:15: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'do_swap_page':
mm/memory.c:2891:17: warning: initialization makes pointer from integer without a cast
mm/memory.c:2948:15: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'do_anonymous_page':
mm/memory.c:3087:16: warning: initialization makes pointer from integer without a cast
mm/memory.c:3108:15: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function '__do_fault':
mm/memory.c:3248:15: warning: initialization makes pointer from integer without a cast
mm/memory.c: In function 'handle_mm_fault':
mm/memory.c:3501:6: warning: assignment makes pointer from integer without a cast
mm/memory.c: In function '__follow_pte':
mm/memory.c:3652:9: warning: initialization makes pointer from integer without a cast

Signed-off-by: WangJing <wangjing@mprc.pku.edu.cn>
---
 arch/unicore64/include/asm/pgtable.h |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/unicore64/include/asm/pgtable.h b/arch/unicore64/include/asm/pgtable.h
index 50ca2c5..f29fc89 100644
--- a/arch/unicore64/include/asm/pgtable.h
+++ b/arch/unicore64/include/asm/pgtable.h
@@ -41,7 +41,12 @@ extern pgd_t swapper_pg_dir[PTRS_PER_PGD];
 #define pmd_offset(pudp, addr)	((pmd_t *)pud_val(*(pudp)) + pmd_index(addr))
 
 #define pte_index(addr)		(((addr) >> PAGE_SHIFT) & (PTRS_PER_PTE - 1))
-#define pte_offset_kernel(pmdp, addr)	({BUG(); 0; })
+static inline pte_t *pte_offset_kernel(pmd_t *pmd, unsigned long address)
+{
+	/* FIXME */
+	BUG();
+	return 0;
+}
 #define pte_offset_map(dir, addr)	pte_offset_kernel((dir), (addr))
 #define pte_unmap(pte)			do { } while (0)
 
-- 
1.7.9.5

